server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java

```diff
@@ -237,7 +246,60 @@ public void defineDeleteMail() {
         });
     }
 
-private Optional<Integer> assertPositiveInteger(Request request, String parameterName) {
+    @PATCH
+    @Path("/{encodedUrl}/mails")
+    @ApiOperation(value = "Deleting all mails in that mailRepository")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            paramType = "query parameter",
+            dataType = "String",
+            example = "?action=clear",
```

Comment:
ok, I realize that I didn't read the updated spec to the end.
What about `DELETE /mailRepositories/{encodedUrl}/mails` ?

Reply:
I also proposed that to @chibenwa and he's telling that that means removing the resource /mails (which is not, it's just emptied), so he prefers this. WDYT?

Reply:
That fact is it doesn't DELETE the repository but the mails, that looks ok to me.

Reply:
Ok